A32NX/ModelBehaviorDefs/A32NX/Interior/A32NX_Interior_Autopilot.xml

```diff
@@ -132,5 +132,308 @@
             <EMISSIVE_CODE_ACTIVE>(L:#PLANE_NAME#_AUTOPILOT_APPR_MODE, bool)</EMISSIVE_CODE_ACTIVE>
         </UseTemplate>
     </Template>
+    
+    	<Template Name="FBW_AUTOPILOT_Knob_SpeedMach_Template">
+		<DefaultTemplateParameters>
+			<ANIM_NAME>AUTOPILOT_Knob_SpeedMach</ANIM_NAME>
+			<ANIM_NAME_PUSH>AUTOPILOT_Knob_SpeedMach_Push</ANIM_NAME_PUSH>
+			<ANIM_NAME_PUSHPULL>AUTOPILOT_Knob_SpeedMach_Push</ANIM_NAME_PUSHPULL>
+			<PART_ID>AUTOPILOT_Knob_SpeedMach</PART_ID>
+			<NODE_ID>AUTOPILOT_Knob_SpeedMach</NODE_ID>
+			<WWISE_EVENT>QNHknob</WWISE_EVENT>
+			<COUNT>36</COUNT>
+			<ANIMTIP_0_ON_CURSOR>TurnLeft</ANIMTIP_0_ON_CURSOR>
+			<ANIMTIP_1_ON_CURSOR>TurnRight</ANIMTIP_1_ON_CURSOR>
+			<ANIMTIP_0>TT:COCKPIT.TOOLTIPS.AUTOPILOT_PANEL_SPEED_DECREASE</ANIMTIP_0>
+			<ANIMTIP_1>TT:COCKPIT.TOOLTIPS.AUTOPILOT_PANEL_SPEED_INCREASE</ANIMTIP_1>
+		</DefaultTemplateParameters>
+		<EditableTemplateParameters>
+			<AddParams>
+				<ANIM_NAME></ANIM_NAME>
+				<ANIM_NAME_PUSHPULL></ANIM_NAME_PUSHPULL>
+			</AddParams>
+		</EditableTemplateParameters>
+		<Component ID="#NODE_ID#" Node="#NODE_ID#">
+			<Switch Param="TYPE">
+				<Case Value="AIRBUS">
+					<UseTemplate Name="ASOBO_GT_Knob_Infinite_PushPull">
+						<ANIM_NAME_KNOB>#ANIM_NAME#</ANIM_NAME_KNOB>
+						<ANTICLOCKWISE_CODE>
+							(L:XMLVAR_AirSpeedIsInMach) if{
+								1 (&gt;K:AP_MACH_VAR_DEC)
+							} els{
+								1 (&gt;K:AP_SPD_VAR_DEC)
+							}
+							(&gt;H:A320_Neo_CDU_AP_DEC_SPEED)
+						</ANTICLOCKWISE_CODE>
+						<CLOCKWISE_CODE>
+							(L:XMLVAR_AirSpeedIsInMach) if{
+								1 (&gt;K:AP_MACH_VAR_INC)
+							} els{
+								1 (&gt;K:AP_SPD_VAR_INC)
+							}
+							(&gt;H:A320_Neo_CDU_AP_INC_SPEED)
+						</CLOCKWISE_CODE>
+						<PULL_CODE>
+							(&gt;H:A320_Neo_CDU_MODE_SELECTED_SPEED)
+						</PULL_CODE>
+						<PUSH_CODE>
+							(&gt;H:A320_Neo_CDU_MODE_MANAGED_SPEED)
+						</PUSH_CODE>
+						<WWISE_EVENT_1>autopilot_knob_push_button_on</WWISE_EVENT_1>
+						<WWISE_EVENT_2>autopilot_knob_push_button_off</WWISE_EVENT_2>
+						<!--<ANIMTIP_0_ON_CURSOR>TurnLeft</ANIMTIP_0_ON_CURSOR>
+						<ANIMTIP_1_ON_CURSOR>TurnRight</ANIMTIP_1_ON_CURSOR> -->
+						<ANIMTIP_2_ON_CURSOR>UpArrow</ANIMTIP_2_ON_CURSOR>
+						<ANIMTIP_3_ON_CURSOR>DownArrow</ANIMTIP_3_ON_CURSOR>
+						<ANIMTIP_0>TT:COCKPIT.TOOLTIPS.AUTOPILOT_PANEL_SPEED_DECREASE</ANIMTIP_0>
+						<ANIMTIP_1>TT:COCKPIT.TOOLTIPS.AUTOPILOT_PANEL_SPEED_INCREASE</ANIMTIP_1>
+						<ANIMTIP_2>TT:COCKPIT.TOOLTIPS.AUTOPILOT_PANEL_SPEED_ENGAGE_MANAGE_MODE</ANIMTIP_2>
+						<ANIMTIP_3>TT:COCKPIT.TOOLTIPS.AUTOPILOT_PANEL_SPEED_ENGAGE_SELECTED_AIRSPEED_MODE</ANIMTIP_3>
+					</UseTemplate>
+				</Case>
+				<Case Value="B747">
```

Comment:
You should remove all the code that is not related to the airbus. It will make the templates much more compact and readable

Reply:
Holland told me not to worry about that and that he will clean up the code later. To be honest I really don't understand templates and it's very confusing for me to edit them without messing anything else up

Reply:
Ah, if he's going to fix it later that's fine.